### ISSUE TITLE
Cant install VPA with Helm Configuration

### ISSUE THREAD

**Issue by nikolic-milan** [kubernetes/autoscaler#5758]
**Which component are you using?**:  cluster-autoscaler

 

**What version of the component are you using?**:

Component version: Latest Version

**What k8s version are you using (`kubectl version`)?**: 1.25

<details><summary><code>kubectl version</code> Output</summary><br><pre>
$ kubectl version 
```Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.2", GitCommit:"5835544ca568b757a8ecae5c153f317e5736700e", GitTreeState:"clean", BuildDate:"2022-09-21T14:33:49Z", GoVersion:"go1.19.1", Compiler:"gc", Platform:"windows/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"24+", GitVersion:"v1.24.10-eks-48e63af", GitCommit:"9176fb99b52f8d5ff73d67fea27f3a638f679f8a", GitTreeState:"clean", BuildDate:"2023-01-24T19:17:48Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"linux/amd64"}````
</pre></details>

**What environment is this in?**: AWS EKS Kubernetes Version 1.24

<!--
If you're using a cloud provider or hardware configuration as your deployment environment let us know here.
-->

**What did you expect to happen?**: I expected a successful installation of autoscaler and VPA

<!--
What behaviour did you expect to see?
-->

**What happened instead?**: I got the fallowing error
```unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "VerticalPodAutoscaler" in version "autoscaling.k8s.io/v1"\n```

<!--
What behaviour did see instead?
-->

**How to reproduce it (as minimally and precisely as possible)**: I used the fallowing code with aws_cdk
```
cluster.add_helm_chart('cluster-autoscaler',
            repository="https://kubernetes.github.io/autoscaler",
            chart="cluster-autoscaler",      
            release="edge-cluster-autoscaler",
            namespace="kube-system",
            values={
                "autoDiscovery": {
                    "clusterName": "EdgeCluster"
                },
                "awsRegion": "eu-central-1",
                "vpa": {
                    "enabled": True,
                    "containerPolicy": {
                        "minAllowed": {
                            "cpu": "20m",
                            "memory": "50Mi"
                        }
                    }
                }
            }        
        )
```

<!--
If possible, provide a recipe for reproducing the error.
A detailed sequence of steps describing what to do to observe the issue is good.
A complete runnable bash shell script is best.
-->

**Anything else we need to know?**: Without the VPA configuration, autoscaler installs.

<!--
Is there anything else you think we should know? Configuration of the component (be careful what you post here if so)? Relevant logs?
-->

**Comment by Shubham82** [comment]
Hi @nikolic-milan 
JFYI: The Helm charts currently install Cluster Autoscaler only not VPA.
https://github.com/kubernetes/autoscaler/tree/master/charts

**Comment by nikolic-milan** [comment]
Hi @Shubham82, I assumed it was since it's documented as such.
https://github.com/kubernetes/autoscaler/tree/master/charts/cluster-autoscaler#verticalpodautoscaler

**Comment by Shubham82** [comment]
I'm sorry, we can install the VPA, but it is for cluster Autoscaler, 
Here is the PR which adds this feature: https://github.com/kubernetes/autoscaler/pull/5558

The error you are facing is due to the version you are using. This feature was merged in CA v1.27 so you should use CA v12.7 and the corresponding k8s version i.e. k8s v1.27.

**Comment by Shubham82** [comment]
Hi @gjtempleton and @voelzmo, is anything I missed here? PTAL!

**Comment by nikolic-milan** [comment]
> I'm sorry, we can install the VPA, but it is for cluster Autoscaler, Here is the PR which adds this feature: #5558
> 
> The error you are facing is due to the version you are using. This feature was merged in CA v1.27 so you should use CA v12.7 and the corresponding k8s version i.e. k8s v1.27.

AWS EKS currently support Kubernetes up to version 1.26.  Would that be a high enough version?
I tried just changing the cluster-autoscaler version to v1.27.0 but I still get the same error.

**Comment by Shubham82** [comment]
I think it is due to the EKS version.
see this: https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler#releases

**Comment by Shubham82** [comment]
> For the CA version, are you sure it's added in v1.27.0?

Yes, it was added in CA v1.27.
Please see the CA-1.27 branch: https://github.com/kubernetes/autoscaler/tree/cluster-autoscaler-release-1.27/charts/cluster-autoscaler
and CA-1.26 branch: https://github.com/kubernetes/autoscaler/tree/cluster-autoscaler-release-1.26/charts/cluster-autoscaler

> The documentation only mentions up to version 1.26.
> https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler#releases

This is only to mention CA compatibility with k8s. i.e. minor release of the CA to the minor release of k8s.

**Comment by gjtempleton** [comment]
Sorry, this is slightly confusing and due to the fact we publish multiple projects from the same repo, it's not related to the CA version.

The CA helm chart now gives the option to install a VPA object to scale the CA as appropriate, however, you need to have already installed the VPA to the cluster separately. We could obviously do with improving the docs around this to make this clearer.

**Comment by Shubham82** [comment]
Thanks, @gjtempleton for the above information.

But to use this feature we should have CA-1.27, am I right?

> The CA helm chart now gives the option to install a VPA object to scale the CA as appropriate, however, you need to have already installed the VPA to the cluster separately. We could obviously do with improving the docs around this to make this clearer.

We can modify/add this thing to VPA content under [README](https://github.com/kubernetes/autoscaler/blob/master/charts/cluster-autoscaler/README.md#verticalpodautoscaler).
Do you think some other place where we can add this?

**Comment by Shubham82** [comment]
> 
> > The CA helm chart now gives the option to install a VPA object to scale the CA as appropriate, however, you need to have already installed the VPA to the cluster separately. We could obviously do with improving the docs around this to make this clearer.
> 
> We can modify/add this thing to VPA content under [README](https://github.com/kubernetes/autoscaler/blob/master/charts/cluster-autoscaler/README.md#verticalpodautoscaler). Do you think some other place where we can add this?

Your thoughts on this?

**Comment by Shubham82** [comment]
Fixed it here: #5763
Thanks
